std/assembly/string.ts

```diff
@@ -380,6 +386,31 @@ export class String {
     );
     return out;
   }
+
+  repeat(count: i32 = 0): String {
+    assert(this !== null);
+    assert(count >= 0);
+    // Most browsers can't handle strings 1 << 28 chars or longer
+    assert(this.length * count <= (1 << 28));
+
+    if (count === 0 || !this.length) return EMPTY;
+    if (count === 1) return this;
+
+    var str    = changetype<String>(this);
+    var result = EMPTY;
+
+    while (count) {
+      if (count & 1) result = result.concat(str);
+      if (count > 1) str    = str.concat(str);
```

Comment:
I wonder if there'd be a benefit in doing this on a lower level, like inlined. Wdyt?

Reply:
Good catch!